src/malli/transform.cljc

```diff
@@ -233,147 +233,146 @@
 
-   :enum {:compile (-infer-child-compiler :encode)}
-   := {:compile (-infer-child-compiler :encode)}
+     'uuid? -any->string
+     #?@(:clj ['uri? -any->string])
 
-   :keyword m/-keyword->string
-   :symbol -any->string
-   :qualified-keyword m/-keyword->string
-   :qualified-symbol -any->string
-   :uuid -any->string
-   ;#?@(:clj [:uri -any->string])
-   ;:bigdec any->string
+     :keyword m/-keyword->string
+     :symbol -any->string
+     :qualified-keyword m/-keyword->string
+     :qualified-symbol -any->string
+     :uuid -any->string
+     ;#?@(:clj [:uri -any->string])
+     ;:bigdec any->string
 
-   'inst? -date->string
-   #?@(:clj ['ratio? -number->double])})
+     'inst? -date->string
+     #?@(:clj ['ratio? -number->double])}))
 
 (defn -string-decoders []
-  (merge
-   (-json-decoders)
-   {'integer? -string->long
-    'int? -string->long
-    'pos-int? -string->long
-    'neg-int? -string->long
-    'nat-int? -string->long
-    'zero? -string->long
-
-    :int -string->long
-    :float -string->float
-    :double -string->double
-    :boolean -string->boolean
-
-    :> -string->long
-    :>= -string->long
-    :< -string->long
-    :<= -string->long
-    :not= -string->long
-
-    'number? -string->double
-    'float? -string->float
-    'double? -string->double
-    #?@(:clj ['rational? -string->double])
-    #?@(:clj ['decimal? -string->decimal])
-
-    'boolean? -string->boolean
-    'false? -string->boolean
-    'true? -string->boolean
-
-    :map-of (-transform-map-keys m/-keyword->string)
-    :vector -sequential->vector}))
+  (-add-child-compilers
+    (merge
+      (-json-decoders)
+      {'integer? -string->long
+       'int? -string->long
+       'pos-int? -string->long
+       'neg-int? -string->long
+       'nat-int? -string->long
+       'zero? -string->long
+
+       :int -string->long
+       :float -string->float
+       :double -string->double
+       :boolean -string->boolean
+
+       :> -string->long
+       :>= -string->long
+       :< -string->long
+       :<= -string->long
+       :not= -string->long
+
+       'number? -string->double
+       'float? -string->float
+       'double? -string->double
+       #?@(:clj ['rational? -string->double])
+       #?@(:clj ['decimal? -string->decimal])
+
+       'boolean? -string->boolean
+       'false? -string->boolean
+       'true? -string->boolean
+
+       :map-of (-transform-map-keys m/-keyword->string)
+       :vector -sequential->vector})))
 
 (defn -string-encoders []
-  (merge
-   (-json-encoders)
-   {'integer? -any->string
-    'int? -any->string
-    'pos-int? -any->string
-    'neg-int? -any->string
-    'nat-int? -any->string
-    'zero? -any->string
-
-    :int -any->string
-    :float -any->string
-    :double -any->string
-    ;:boolean -any->string
-
-    :> -any->string
-    :>= -any->string
-    :< -any->string
-    :<= -any->string
-    :not= -any->string
-
-    'float -any->string
-    'double -any->string}))
+  (-add-child-compilers
```

Comment:
Hmm, so `-add-child-compilers` will overwrite the `:=` and `:enum` keys from `-json-encoders`. I guess that works, but somehow I feel the composability of the encoders is decreased.

Reply:
I pushed a change to pull out the "base" encoders and decoders into separate functions. It doesn't change anything functionally, but gets rid of the overwriting you mentioned, so it's at least more straightforward maybe? Let me know what you think.